### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by sohambaghela-git**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -14,5 +14,12 @@ def authenticate_request
 	def current_user
 		@current_user.present?
 	end
+	
+	# To create a cart for User
+	def create_unique_cart
```
Comment (vlovanshi) [inline app/controllers/application_controller.rb]: Create cart on after_save

File: app/controllers/cart_controller.rb [hunk 2]
```diff
@@ -1,19 +1,9 @@
...
-    cart.save
-    render json: { cart: cart, message: " product Added in cart " }
-  end
-
-  private 
-
-  def cart_params
-    params.require(:cart).permit!
+    cart = @current_user.cart
+    cart_item = CartItem.where(cart_id: cart.id)
```
Comment (vlovanshi) [inline app/controllers/cart_controller.rb]: cart_items

File: app/controllers/cart_items_controller.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  def index
+  end
+
+  def create 
+    cart_item = CartItem.new(cart_item_params)
+    cart_item.user_id = @current_user.id
+  
+    if cart_item.save
+      render json: {message: "product added into cart ", status: :ok}
```
Comment (vlovanshi) [inline app/controllers/cart_items_controller.rb]: Message first letter should be capital

File: app/controllers/products_controller.rb [hunk 4]
```diff
@@ -13,7 +14,7 @@ def show
 	def create
 		product = Product.new(product_params)
 		product.user_id = @current_user.id
-		if $current_user.role == ("seller")
+		if @current_user.role == ("seller")
```
Comment (vlovanshi) [inline app/controllers/products_controller.rb]: create method in user model

def seller?
  self.role == 'seller'
end

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,27 +1,30 @@
...
+		if (params[:id].to_i) == (@current_user.id)
 			render json: @user
 		else
-			render json: " Sorry You can not "
+			render json: " Sorry You can not access this url"
 		end
 	end
 
 	def create
 		@user = User.new(user_params)
```
Comment (vlovanshi) [inline app/controllers/users_controller.rb]: Don't create instance variable until it is required.

File: app/models/user.rb [hunk 6]
```diff
@@ -1,6 +1,10 @@
 class User < ApplicationRecord
   has_secure_password
   has_many :products
+	has_one :cart
+	has_many :cart_items
+
+	# User Register Validation
 	validates :first_name, presence: true
```
Comment (vlovanshi) [inline app/models/user.rb]: Add presence true validation in single line

**Comment by sohambaghela-git** [comment]
I have fixed all bugs

**Pull request merged**